peps/pep-0817.rst

```diff
@@ -0,0 +1,2494 @@
+upgraded packages being reverted to the default variant on subsequent upgrades.
+
+
+Package names as variants
+'''''''''''''''''''''''''
+
+Packages such as `XGBoost
+<https://xgboost.readthedocs.io/en/stable/install.html>`__ use different
+package names to approximate variants:
+
+.. code:: bash
+
+    pip install xgboost      # NVIDIA GPU variant
+    pip install xgboost-cpu  # CPU-only variant
+
+Maintainers of other software cannot express that they depend on either
+of the available variants being selected. They need to
+either depend on a specific variant, provide multiple alternative
+dependency sets using extras, or even publish their own software using
+multiple package names matching upstream variants.
+
+Commonly, these packages install overlapping files. Since Python
+packaging does not support expressing that two packages are mutually
+exclusive, installers can install both of them to the same environment,
+with the package installed second overwriting files from the one
+installed first. This leads to runtime errors, and
+the possibility of incidentally switching between variants depending on
+the way package upgrades are ordered.
+
+An additional limitation of this approach is that publishing a new release
+synchronously across multiple package names is not currently possible.
+:pep:`694` proposes adding such a mechanism for multiple wheels within a
+single package, but extending it to multiple packages is not a goal.
+
+**Induced Security Risk:** proliferation of suffixed variant packages
+leads users to expect these suffixes in other packages, making name
+squatting much easier. For example, one could create a malicious
+``numpy-cuda`` package that users will be lead to believe it’s a CUDA
+variant of NumPy.
+
+`CuPy <https://cupy.dev/>`_ had to build
+a total of 52 different packages - all with different names - which
+clearly highlights the limit of such an approach. End users need to
+carefully read the CuPy installation documentation to figure out
+which package they need. Continuously having to create new PyPI
+packages, request increasing limits and keep the build infrastructure
+and documentation in sync, requires significant effort from software
+maintainers.
+
+.. code:: text
+
+    cupy-cuda100 cupy-cuda101 cupy-cuda102 cupy-cuda110 cupy-cuda111 cupy-cuda112 cupy-cuda113 cupy-cuda114 cupy-cuda115
+    cupy-cuda116 cupy-cuda117 cupy-cuda118 cupy-cuda119 cupy-cuda11x cupy-cuda120 cupy-cuda121 cupy-cuda122 cupy-cuda123
+    cupy-cuda124 cupy-cuda125 cupy-cuda126 cupy-cuda127 cupy-cuda128 cupy-cuda129 cupy-cuda12x cupy-cuda13x cupy-cuda70
+    cupy-cuda75 cupy-cuda80 cupy-cuda90 cupy-cuda91 cupy-cuda92 cupy-rocm-4-0 cupy-rocm-4-1 cupy-rocm-4-2 cupy-rocm-4-3
+    cupy-rocm-4-4 cupy-rocm-4-5 cupy-rocm-5-0 cupy-rocm-5-1 cupy-rocm-5-2 cupy-rocm-5-3 cupy-rocm-5-4 cupy-rocm-5-5
+    cupy-rocm-5-6 cupy-rocm-5-7 cupy-rocm-5-8 cupy-rocm-5-9 cupy-rocm-6-0 cupy-rocm-6-1 cupy-rocm-6-2 cupy-rocm-6-3
```

Comment:
If it's useful to keep all 52 names (shock factor!), please wrap to 70-ish characters, as literal blocks aren't reflowed on smaller screens.